src/scripts/main.js

```diff
@@ -1,3 +1,70 @@
+/* eslint-disable no-console */
+/* eslint-disable no-undef */
+/* eslint-disable operator-linebreak */
+/* eslint-disable space-before-function-paren */
+
 'use strict';
 
-// write code here
+const BASE_URL =
+  'https://mate-academy.github.io/phone-catalogue-static/api/phones';
+
+const request = (url) => {
+  return new Promise((resolve) => {
+    fetch(url)
+      .then((res) => {
+        if (res.status < 400) {
+          return res.json();
+        } else {
+          throw res;
+        }
+      })
+      .then((res) => resolve(res))
+      .catch((err) =>
+        setTimeout(() => {
```

Comment:
```suggestion
  return new Promise((resolve, reject) => {
    setTimeout(reject, 5000);
    
    fetch(url)
      .then((res) => {
        if (res.status < 400) {
          return res.json();
        } else {
          throw res;
        }
      })
      .then((res) => resolve(res))
```
і тоді на своєму request catch помилки